e2e/tests/regression/test_v2.robot

```diff
@@ -50,3 +52,23 @@ Degraded Volume Replica Rebuilding
         And Wait for deployment 0 pods stable
         Then Check deployment 0 data in file data.txt is intact
     END
+
+V2 Volume Should Block Trim When Volume Is Degraded
+    Given Set setting auto-salvage to true
+    And Create storageclass longhorn-test with    dataEngine=v2
+    And Create persistentvolumeclaim 0 using RWO volume with longhorn-test storageclass
+    And Create deployment 0 with persistentvolumeclaim 0
+
+    FOR    ${i}    IN RANGE    ${LOOP_COUNT}
+        And Keep writing data to pod of deployment 0
+
+        When Restart cluster
+        And Wait for longhorn ready
+        And Wait for volume of deployment 0 attached and degraded
+        Then Trim deployment 0 volume should fail
+
+        When Wait for workloads pods stable
```

Comment:
Could we add step to wait for volume healthy before this step? because if I understand correctly, workload pod stable running not guarantee all replica running, beside this, no other questions, thank you.